### ISSUE TITLE
[BUG] - Select scrolls page and immediately closes popover on click

### ISSUE THREAD

**Issue by hyperborea** [heroui-inc/heroui#3571]
### NextUI Version

2.4.6

### Describe the bug

Clicking a `<Select/>` trigger, when the currently selected key is sufficiently far down the list, causes the page to scroll and the popover not to show (or more precisely, close it immediately as soon as it's opened). It sometimes also seems to select another key.

When clicking the trigger again (at the new scroll position), the dropdown opens correctly. This assumes the new "stable" scroll position could actually be reached. I first encountered the issue on iOS Safari on my project where the container is `position: fixed`, so it just keeps scrolling when clicking the trigger.

Note that the page will scroll down when you're at the top. If you scroll down the page enough and click the select it will scroll up instead to get the component to the same "stable" scroll position.

This issue is driving me crazy, maybe someone else has an idea what's up.

### Your Example Website or App

https://stackblitz.com/edit/nextui-select-scroll-bug?file=src%2FApp.tsx

### Steps to Reproduce the Bug or Issue

1. click select and pick "7" or higher
2. click select and observe bug

Note: the issue ONLY appears on "7" or higher, any items further up the list are fine. In other words, the item needs to partially or fully below the scroll threshold of the popover / scroll shadow.

### Expected behavior

The dropdown should open normally and the page should not scroll up.

### Operating System Version

macOS

### Browser

Chrome

**Comment by hyperborea** [comment]
It would appear to me like there's possibly two issues:
1) something causes the page to scroll
2) when the popover overlaps the component, the same click that opens the popover also selects whatever item is at the cursor

**Comment by vavilondev** [comment]
I have the same issue. And had the same in previous versions.
Actually, you have that issue on your next ui page.

https://github.com/user-attachments/assets/b82322d9-7448-4270-9577-a218f384b097

**Comment by hyperborea** [comment]
That's right, I've double checked and the problem seems to have been there since the component was introduced in `2.1.0`. This is really quite problematic - at best it's bad UX (page is scrolled unnecessarily and requires a second click to open the component), at worst it prevents it from functioning at all (e.g. when positioned as `sticky` or `fixed`).

**Comment by n-d-r-d-g** [comment]
Same issue here. I agree with @hyperborea. I'm using the Select component in a sticky header and I need it to be a controlled component.

I've noticed that whenever i use `defaultSelectedKeys` or selectedKeys, I cannot open the Select unless I scroll to the top of the page. If I scroll even just a few pixels down and click the Select, it opens and closes immediately.

I'm using version `2.4.6`.

**Comment by quiloos39** [comment]
Bump any updates having similar issues

**Comment by j-h-scheufen** [comment]
I'm facing the exact same problem using "@nextui-org/select": "^2.2.7".

**Comment by iamfrisbee** [comment]
I have had a hard time recreating it outside the app I'm working on but here is how you might see it.
1. https://4mjhgj-3000.csb.app/
2. Open chrome dev tools
3. Turn on the device toolbar
4. Choose the smallest device
5 Scroll to the very last select and try to update it

**Comment by zackteng** [comment]
I'm having this problem too. using "@nextui-org/select": "^2.2.7".
Does anyone know how to quickly fix this in my own project code?

**Comment by dragos99** [comment]
Same issue here, this is impacting UX to the point where they become frustrated. Unfortunately this makes NextUI an unusable library, which is a shame, because otherwise it's great.

**Comment by L1mit1337** [comment]
same too,Is the bug fixed? I'm still triggering this bug

**Comment by datkira** [comment]
Hope this bug fixed soon!

**Comment by JonathanTylerWeber** [comment]
having the same issue

**Comment by JonathanTylerWeber** [comment]
I'm not sure if this is the right way to do it, but it fixed my problem:
```
<Select
  className="max-w-xs"
  // other props
  popoverProps={{
    classNames: {
      content: "max-h-52",
    },
  }}
>...</Select>
```

**Comment by n-d-r-d-g** [comment]
> I'm not sure if this is the right way to do it, but it fixed my problem:
> 
> ```
> <Select
>   className="max-w-xs"
>   // other props
>   popoverProps={{
>     classNames: {
>       content: "max-h-52",
>     },
>   }}
> >...</Select>
> ```

Unfortunately, this doesn't fix the issue on my end.

**Comment by seanghods** [comment]
Broken for me too, shame

**Comment by zackteng** [comment]
Any progress on this bug? It's been months.

**Comment by yurkimus** [comment]
This has happened to me randomly this evening. Everything was nice with a feature during the beta release and then the problem popped-out from nowhere.

I have figured out either the `selectedKeys` or `defaultSelectedKeys` are somehow related to the issue. If I remove the property - everything works fine.

At the moment our team is blocked with delivering a beautiful UI update for the application. That's crazy 😟
